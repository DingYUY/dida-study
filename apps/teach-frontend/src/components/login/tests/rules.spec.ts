import { describe, expect, it } from 'vitest'
import { createConfirmPasswordRule, createPasswordRule, createUsernameRule } from '../rules'

describe('rules', () => {
  it('create name rule', () => {
    expect(createUsernameRule()).toMatchSnapshot()
  })
  it('create password rule', () => {
    expect(createPasswordRule()).toMatchSnapshot()
  })
  it('create confirm password rule', () => {
    expect(createConfirmPasswordRule({ password: '' })).toMatchSnapshot()
  })
})
