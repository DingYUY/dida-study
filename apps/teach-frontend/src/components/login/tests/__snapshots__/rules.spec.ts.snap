// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rules > create confirm password rule 1`] = `
[
  {
    "message": "请再次输入密码",
    "required": true,
    "trigger": [
      "input",
      "blur",
    ],
  },
  {
    "message": "两次密码输入不一致",
    "trigger": [
      "blur",
      "password-input",
    ],
    "validator": [Function],
  },
]
`;

exports[`rules > create name rule 1`] = `
[
  {
    "message": "请输入帐号",
    "required": true,
    "trigger": "blur",
  },
  {
    "message": "长度要大于 6 小于 25",
    "trigger": "blur",
    "validator": [Function],
  },
]
`;

exports[`rules > create password rule 1`] = `
[
  {
    "message": "请输入密码",
    "required": true,
    "trigger": "blur",
  },
  {
    "message": "长度要大于 6 小于 30",
    "trigger": "blur",
    "validator": [Function],
  },
]
`;
